src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -290,6 +292,12 @@ private void OnGetHeadersMessageReceived(GetBlockByIndexPayload payload)
             EnqueueMessage(Message.Create(MessageCommand.Headers, HeadersPayload.Create(headers.ToArray())));
         }
 
+        private void OnHeadersMessageReceived(HeadersPayload payload)
+        {
+            if (payload.Headers.Length == 0) return;
```

Comment:
Move this check to `Header.Deserialize()`?

Reply:
Fixed